File: Cocktail/src/main/kotlin/org/programmers/cocktail/search/service/CocktailLikesMapper.java
package org.programmers.cocktail.search.service;

import org.modelmapper.ModelMapper;
import org.modelmapper.PropertyMap;
import org.programmers.cocktail.entity.CocktailLikes;
import org.programmers.cocktail.search.dto.CocktailLikesTO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class CocktailLikesMapper {
    private ModelMapper modelMapper;

    @Autowired
    public CocktailLikesMapper(ModelMapper modelMapper) {
        this.modelMapper = modelMapper;

        // CocktailLikes -> CocktailLikesTO 변환 매핑
        modelMapper.addMappings(new PropertyMap<CocktailLikes, CocktailLikesTO>() {
            @Override
            protected void configure() {
                map().setUserId(source.getUsers().getId());
                map().setCocktailId(source.getCocktails().getId());
            }
        });

        // CocktailLikesTO -> CocktailLikes 변환 매핑
        modelMapper.addMappings(new PropertyMap<CocktailLikesTO, CocktailLikes>() {
            @Override
            protected void configure() {
                map().getUsers().setId(source.getUserId());
                map().getCocktails().setId(source.getCocktailId());
            }
        });
    }

    public CocktailLikesTO convertToCocktailsLikesTO(CocktailLikes cocktailLikes) {
        return modelMapper.map(cocktailLikes, CocktailLikesTO.class);
    }

    public CocktailLikes convertToCocktailLikes(CocktailLikesTO cocktailLikesTO){
        return modelMapper.map(cocktailLikesTO, CocktailLikes.class);
    }
}

File: Cocktail/src/main/kotlin/org/programmers/cocktail/search/service/CocktailLikesMapper.kt
package org.programmers.cocktail.search.service

import org.programmers.cocktail.entity.CocktailLikes
import org.programmers.cocktail.search.dto.CocktailLikesTO
import org.springframework.stereotype.Service

@Service
object CocktailLikesMapper {

    // CocktailLikes -> CocktailLikesTO 변환
    fun convertToCocktailsLikesTO(cocktailLikes: CocktailLikes?): CocktailLikesTO {
        return CocktailLikesTO(
            userId = cocktailLikes?.users?.id,    // users 객체에서 id를 가져와서 userId로 설정
            cocktailId = cocktailLikes?.cocktails?.id // cocktails 객체에서 id를 가져와서 cocktailId로 설정
        )
    }

    // CocktailLikesTO -> CocktailLikes 변환
    fun convertToCocktailLikes(cocktailLikesTO: CocktailLikesTO?): CocktailLikes {

        var cocktailLikes : CocktailLikes = CocktailLikes()
        cocktailLikes.users.id = cocktailLikesTO?.userId
        cocktailLikes.cocktails.id = cocktailLikesTO?.cocktailId

        return cocktailLikes
    }
}
